nerfstudio/fields/sdf_field.py

```diff
@@ -0,0 +1,451 @@
+    """
+
+    config: SDFFieldConfig
+
+    def __init__(
+        self,
+        config: SDFFieldConfig,
+        aabb: TensorType[2, 3],
+        num_images: int,
+        use_average_appearance_embedding: bool = False,
+        spatial_distortion: Optional[SpatialDistortion] = None,
+    ) -> None:
+        super().__init__()
+        self.config = config
+
+        self.aabb = Parameter(aabb, requires_grad=False)
+
+        self.spatial_distortion = spatial_distortion
+        self.num_images = num_images
+
+        self.embedding_appearance = Embedding(self.num_images, self.config.appearance_embedding_dim)
+        self.use_average_appearance_embedding = use_average_appearance_embedding
+        self.use_grid_feature = self.config.use_grid_feature
+        self.divide_factor = self.config.divide_factor
+
+        growth_factor = np.exp((np.log(config.max_res) - np.log(config.base_res)) / (config.num_levels - 1))
+
+        if self.config.encoding_type == "hash":
+            # feature encoding
+            self.encoding = tcnn.Encoding(
+                n_input_dims=3,
+                encoding_config={
+                    "otype": "HashGrid" if config.use_hash else "DenseGrid",
+                    "n_levels": config.num_levels,
+                    "n_features_per_level": config.features_per_level,
+                    "log2_hashmap_size": config.log2_hashmap_size,
+                    "base_resolution": config.base_res,
+                    "per_level_scale": growth_factor,
+                    "interpolation": "Smoothstep" if config.smoothstep else "Linear",
+                },
+            )
+
+        # we concat inputs position ourselves
+        self.position_encoding = NeRFEncoding(
+            in_dim=3, num_frequencies=6, min_freq_exp=0.0, max_freq_exp=5.0, include_input=False
+        )
+
+        self.direction_encoding = NeRFEncoding(
+            in_dim=3, num_frequencies=4, min_freq_exp=0.0, max_freq_exp=3.0, include_input=True
+        )
+
+        # initialize geometric network
+        self.initialize_geo_layers()
+
+        # deviation_network to compute alpha from sdf from NeuS
+        self.deviation_network = LearnedVariance(init_val=self.config.beta_init)
+
+        # color network
+        dims = [self.config.hidden_dim_color for _ in range(self.config.num_layers_color)]
+        # point, view_direction, normal, feature, embedding
+        in_dim = (
+            3
+            + self.direction_encoding.get_out_dim()
+            + 3
+            + self.config.geo_feat_dim
+            + self.embedding_appearance.get_out_dim()
+        )
+        dims = [in_dim] + dims + [3]
+        self.num_layers_color = len(dims)
+
+        for l in range(0, self.num_layers_color - 1):
+            out_dim = dims[l + 1]
+            lin = nn.Linear(dims[l], out_dim)
+
+            if self.config.weight_norm:
+                lin = nn.utils.weight_norm(lin)
+            # print("=======", lin.weight.shape)
```

Comment:
Delete. Can you go through and see if there are other commented out code.

Reply:
Apologies thought I had already gotten it. Will go through again